STL_Extension/include/CGAL/result_of.h

```diff
@@ -21,42 +21,49 @@
 #include <CGAL/config.h>
 #include <CGAL/disable_warnings.h>
 
-// Address the warning C4003: not enough actual parameters for macro 'BOOST_PP_SEQ_DETAIL_IS_NOT_EMPTY'
-// result_of.hpp includes files from boost/preprocessor
-// This concerns boost 1_65_1
-#if defined(BOOST_MSVC)
-#  pragma warning(push)
-#  pragma warning(disable: 4003)
-#endif
-#include <boost/utility/result_of.hpp>
-#if defined(BOOST_MSVC)
-#  pragma warning(pop)
-#endif
+#if CGAL_CXX11
 
-#include <boost/version.hpp>
+  #include <type_traits>
 
-namespace CGAL{
+#else // C++<11
 
-// Even if for now we use the tr1 result_of implementation, we use the cpp11
-// namespace since in the future, that's the decltype version that will be used
-namespace cpp11{
-
-template<typename F>
-struct result_of
-{
-  // from boost 1.44 release note https://www.boost.org/users/history/version_1_44_0.html :
-  //    New template boost::tr1_result_of that implements the TR1 ResultOf protocol even if boost::result_of uses the C++0x decltype-based implementation.
-  #if BOOST_VERSION < 104400
-  typedef typename boost::result_of<F>::type type;
-  #else
-  typedef typename boost::tr1_result_of<F>::type type;
+  // Address the warning C4003: not enough actual parameters for macro 'BOOST_PP_SEQ_DETAIL_IS_NOT_EMPTY'
+  // result_of.hpp includes files from boost/preprocessor
+  // This concerns boost 1_65_1
+  #if defined(BOOST_MSVC)
+    #pragma warning(push)
+    #pragma warning(disable: 4003)
+  #endif
+    #include <boost/utility/result_of.hpp>
+  #if defined(BOOST_MSVC)
+    #pragma warning(pop)
   #endif
-};
+  #include <boost/version.hpp>
+
+#endif // end C++<11
+
+namespace CGAL {
+namespace cpp11 {
+
+#if CGAL_CXX20
 
-}
+  template<typename Signature> class result_of;
+  template<typename F, typename... Args>
+  class result_of<F(Args...)> : public std::invoke_result<F, Args...> { };
+
+#elif CGAL_CXX11
+
+  using std::result_of;
+
+#else // C++<11
+
+  using boost::result_of;
+
+#endif
 
```

Comment:
This produces warnings in Visual Studio, as result_of was deprecated in c++17 : https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-5.3-Ic-58/Generator_Demo/TestReport_afabri_x64_Cygwin-Windows10_MSVC2017-Debug-64bits.gz

Reply:
I guess it should also be conditioned with cxx < 17 or something

Reply:
also errors : https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-5.3-Ic-58/Alpha_shapes_2_Examples/TestReport_lrineau_Fedora.gz

Reply:
Isn't there already `#if CGAL_CXX20` above? Maybe that doesn't match VS with c++latest and `__cpp_lib_is_invocable>=201703L` would work better?